lib/testutils/client.go

```diff
@@ -77,18 +63,88 @@ func CreateNewIPPool(c client.Client, poolSubnet string, ipip, natOut, ipam bool
 	if err != nil {
 		log.Printf("Error creating pool: %s\n", err)
 	}
+}
+
+func CreateClient(config api.CalicoAPIConfig) *client.Client {
+	c, err := client.New(config)
+
+	if err != nil {
+		panic(err)
+	}
 
+	return c
 }
 
-// CleanIPPools removes all IP pool configuration from the datastore.
 func CleanIPPools(c *client.Client) {
 	if pools, err := c.IPPools().List(api.IPPoolMetadata{}); err == nil {
 		for _, pool := range pools.Items {
 			if err := c.IPPools().Delete(pool.Metadata); err != nil {
 				panic(err)
 			}
 		}
-	} else {
+	}
+}
+
+func CleanDatastore(config api.CalicoAPIConfig) {
+	var err error
+
+	log.Println(fmt.Sprintf("Cleaning datastore: %v", config.Spec.DatastoreType))
+
+	switch config.Spec.DatastoreType {
+	case api.EtcdV2:
+		cfg := etcdclient.Config{
+			Endpoints: []string{config.Spec.EtcdScheme + "://" + config.Spec.EtcdAuthority},
+		}
+		if config.Spec.EtcdEndpoints != "" {
+			cfg = etcdclient.Config{
+				Endpoints: strings.Split(config.Spec.EtcdEndpoints, ","),
+			}
+		}
+		if c, err := etcdclient.New(cfg); c != nil {
+			kapi := etcdclient.NewKeysAPI(c)
+			_, err = kapi.Delete(context.Background(), "/calico", &etcdclient.DeleteOptions{Dir: true, Recursive: true})
+		} else {
+			log.Errorf("Can't create etcd backend %v", err)
+		}
+	default:
+		err = errors.New(fmt.Sprintf("Unknown datastore type: %v", config.Spec.DatastoreType))
+	}
+
+	if err != nil {
 		panic(err)
 	}
 }
+
+// CleanBackend is a utility function to wipe clean "/calico" recursively from backend.
+func CreateCleanClient(config api.CalicoAPIConfig) *client.Client {
+	CleanDatastore(config)
+
+	return CreateClient(config)
+}
+
+// DumpBackend prints out a recursive dump of the contents of backend.
+func DumpBackend(config api.CalicoAPIConfig) error {
```

Comment:
Since we're tidying this up - I think renaming it to DumpDatastore might be better.  WDYT?